### PR TITLE
Emit warning trying to read ECSV file without PyYAML

### LINKED ISSUE #7564: Request for better warning: PyYaml package is required for Unit support is astropy.io.ascii.read
The problem occurs with the following code:
```python
from astropy.io import ascii
t = ascii.read('WASP43_LDCs_A100.txt')
assert t["wavelength"].unit != None
```

The data file:
[WASP43_LDCs_A100.txt](https://github.com/astropy/astropy/files/2101773/WASP43_LDCs_A100.txt)

If the module pyyaml is not installed, Unit are not working properly and instead of having "microns", I simply get "None" and my code crashes. 

Thanks to astrofrog (from the astropy Slack), we've successfully linked my crash problem with this package missing. 

Since this package is not automatically installed with astropy, and we currently don't get any warning that we might need it. 

Could something be done to:

1. warn user, 
2. add a dependency to astropy (either in ascii.read or in astropy.unit) ?

I would personnally be in favor of simply adding "pyyaml" to the requirements.txt, but I don't know what is your policy in the Astropy Team, and maybe another solution might be preferred.

### PR DESCRIPTION
**Pull request by taldcroft**
Fixes #7564.

### CODE REVIEW

File: astropy/io/ascii/ecsv.py [hunk 1]
```diff
@@ -124,6 +125,13 @@ def get_cols(self, lines):
 
         try:
             header = meta.get_header_from_yaml(lines)
+        except ImportError as exc:
+            if 'PyYAML package is required' in str(exc):
+                warnings.warn("file looks like ECSV format but PyYAML is not installed "
+                              "so it cannot be parsed as ECSV",
+                              AstropyWarning)
+            raise core.InconsistentTableError('unable to parse yaml in meta header'
```
Comment (taldcroft) [inline astropy/io/ascii/ecsv.py]: I would have preferred to have this raise the `ImportError`, but that doesn't get caught in guessing.  Hence the compromise of `InconsistentTableError` with text about the import.

File: astropy/io/ascii/ui.py [hunk 2]
```diff
@@ -608,8 +608,7 @@ def _get_guess_kwargs_list(read_kwargs):
 
     # Start with ECSV because an ECSV file will be read by Basic.  This format
     # has very specific header requirements and fails out quickly.
-    if HAS_YAML:
-        guess_kwargs_list.append(dict(Reader=ecsv.Ecsv))
+    guess_kwargs_list.append(dict(Reader=ecsv.Ecsv))
```
Comment (taldcroft) [inline astropy/io/ascii/ui.py]: In order to handle this, we need to always try ECSV even if PyYAML is not installed.  Guessing just got a little bit slower (but not too bad).

File: CHANGES.rst [hunk 3]
```diff
@@ -56,6 +56,10 @@ astropy.io.ascii
   All guesses thus removed from ``filtered_guess_kwargs`` are now listed as
   "Disabled" at the beginning of the trace. [#5578]
 
+- Emit a warning when reading an ECSV file without specifying the ``format``
+  and without PyYAML installed.  Previously this silently fell through to
+  parsing as a basic format file and the file metadata was lost. [#7579]
```
Comment (pllim) [inline CHANGES.rst]: PR number is 7580, please update. Thanks.

Comment (taldcroft) [reply CHANGES.rst]: Done.

**Pull request merged**